### PR TITLE
New styling and icons. Bug fix. DQA

### PR DESCRIPTION
**Pull request by Camwyn**
[EXT-241]

[EXT-241]: https://moderntribe.atlassian.net/browse/EXT-241

This PR adds:

- A new view icon.
- A few small style thingies from the last DQA.

and fixes:
- The constants containing "COMPACT" - as it's now called "SUMMARY" TYVM.
- A bug where a multiday event will not show on a day (other than it's start date) where there are no other events.
  - 📷 http://p.tri.be/xd9mSA
- This also adds auto-enabling the view on plugin activation. 
  - 🎥 http://p.tri.be/oo21Ux

**DQA update:** 
:camera: http://p.tri.be/D6AZCf
:camera: http://p.tri.be/LRBxsc

### CODE REVIEW

**Comment by Camwyn** [comment]
Marking this "ready for review since I did change a couple small php things. 
Can always create a DQA PR as needed.

File: src/Tribe/Plugin.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ public function register() {
 		$this->container->singleton( static::class, $this );
 		$this->container->singleton( 'extension.summary_view', $this );
 		$this->container->singleton( 'extension.summary_view.plugin', $this );
-		$this->container->register( PUE::class );
+		//$this->container->register( PUE::class );
```
Comment (Camwyn) [inline src/Tribe/Plugin.php]: Not sure this should be in here? licensing?

File: src/Tribe/Views/Summary_View.php [hunk 2]
```diff
@@ -191,7 +191,7 @@ protected function get_events_that_start_and_end_around( $first_date, $last_date
 	 * @param \WP_Post $earliest_event First event in result set.
 	 * @param array $exclude_ids Event IDs in result set.
 	 *
-	 * @return array
+	 * @return \WP_Post
```
Comment (Camwyn) [inline src/Tribe/Views/Summary_View.php:194]: This was throwing param mismatch warnings

File: src/views/v2/components/icons/summary.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * View: Summary Icon
```
Comment (Camwyn) [inline src/views/v2/components/icons/summary.php:3]: MCR is supposed to provide a tweaked version of this.

File: src/views/v2/summary/event/date.php [hunk 4]
```diff
@@ -23,21 +23,22 @@
...
-<div class="tribe-events-calendar-summary__event-datetime-wrapper tribe-common-b2">
+<div class="tribe-events-calendar-summary__event-datetime-wrapper tribe-common-b3">
 	<?php $this->template( 'summary/event/date/featured' ); ?>
+	<?php $this->template( 'summary/event/date/virtual' ); ?>
 	<time class="tribe-events-calendar-summary__event-datetime" datetime="<?php echo esc_attr( $formatted_start_date ); ?>" title="<?php echo $event->start_date . ' :: ' . $event->end_date; ?>">
 		<?php if ( ! $event->multiday ) : ?>
 			<span class="tribe-event-date-start"><?php echo esc_html( $event->summary_view->start_time ); ?></span> -
 			<span class="tribe-event-date-end"><?php echo esc_html( $event->summary_view->end_time ); ?></span>
-		<?php elseif ( $event->summary_view->start_date === $formatted_group_date ) : ?>
+		<?php elseif ( ! empty( $event->summary_view ) && $event->summary_view->start_date === $formatted_group_date ) : ?>
```
Comment (Camwyn) [inline src/views/v2/summary/event/date.php:32]: As discussed - this sometimes throws a warning, so we now check `empty()` before checking the value.

File: plugin.php [hunk 5]
```diff
@@ -72,3 +72,17 @@ function tribe_extension_summary_view() {
...
+register_activation_hook( __FILE__, 'tribe_extension_summary_view_activation' );
+
+/**
+ * Enables the view on plugin activation.
+ *
+ * @since 1.0.0
+ */
+function tribe_extension_summary_view_activation() {
+	$enabled_views   = tribe_get_option('tribeEnableViews');
+	$enabled_views[] = 'summary';
```
Comment (borkweb) [inline plugin.php:90]:
```suggestion
	if ( in_array( 'summary', $enabled_views ) ) {
		return;
	}
	
	$enabled_views[] = 'summary';
```

File: src/Tribe/Plugin.php [hunk 6]
```diff
@@ -144,4 +138,11 @@ protected function register_plugin_dependencies() {
 		$this->container->singleton( Plugin_Register::class, $plugin_register );
 		$this->container->singleton( 'extension.summary_view', $plugin_register );
 	}
+
+	protected function plugin_activated() {
```
Comment (borkweb) [inline src/Tribe/Plugin.php]: Add a docblock here

**Review by borkweb — CHANGES_REQUESTED** [review]
This looks pretty great! Just a couple of tweaks.

File: src/Tribe/Plugin.php (hunk 6 above)
Comment (Camwyn) [reply src/Tribe/Plugin.php]: Whups - removing this, it's unused.

File: plugin.php (hunk 5 above)
Comment (Camwyn) [reply plugin.php:90]: corrected (above marked line)

**Review by borkweb — APPROVED** [review]
:+1:

**Pull request merged**